streaming/base/shared.py

```diff
@@ -268,3 +272,79 @@ def cleanup(self):
             pass
         finally:
             resource_tracker.unregister = original_rtracker_unreg
+
+
+def get_shm_prefix(my_locals: List[str], world: World) -> Tuple[str, SharedMemory]:
+    """Register or lookup our shared memory prefix.
+
+    Args:
+        my_locals (List[str]): Local working dir of each stream, relative to /. We need to verify
+            that there is no overlap with any other currently running StreamingDataset.
+        world (World): Our place in the world.
+
+    Returns:
+        Tuple[str, CreateSharedMemory]: Shared memory prefix and object. The name is required to be
+            very short for Mac OSX.
+    """
+    # Local leader goes first, checking and registering.
+    if world.is_local_leader:
+        # Local leader walks the existing shm prefixes starting from zero, verifying that there is
+        # no local working directory overlap.
+        for prefix_int in range(10**6):
+            prefix = f'{prefix_int:06}'
+            name = f'{prefix}_locals'
+            try:
+                shm = CreateSharedMemory(name, False).shm
+            except:
+                break
+            size = np.frombuffer(shm.buf[:4], np.int32)[0]
+            data = bytes(shm.buf[4:size])
+            text = data.decode('utf-8')
+            their_locals = text.split('\0')
+            both = set(my_locals) & set(their_locals)
+            if both:
+                raise ValueError(f'Reused local working directory: {sorted(my_locals)} vs ' +
```

Comment:
What is the suggestion here for the user? What they need to do to not see the same error message again ?

Reply:
They need to not reuse a local directory